sympy/functions/special/tests/test_error_functions.py

```diff
@@ -613,6 +613,13 @@ def test_si():
     assert tn_arg(Si)
     assert tn_arg(Shi)
 
+    assert Si(x)._eval_as_leading_term(x) == x
```

Comment:
there is no the case `arg0 is S.nan`. Or I'm wrong?

Reply:
Well yes there is no case where arg0 is `S.nan` but then you don't encounter those on a regular basis so that's fine. If know of a mathematical function whose leading term when substituted with 0 for x gives S.Nan that would help. But in gen if you want an explanation.

`exp(1/x).subs(x, 0)` gives us `S.Nan` and this works perfectly 
```
>>> limit(Si(exp(1/x)), x, 0, '+')
pi/2
>>> limit(Si(exp(1/x)), x, 0, '-')
0
```